tests/unit-core/test-stringbuilder.c

```diff
@@ -0,0 +1,244 @@
+    ecma_deref_ecma_string (result_p);
+    ecma_deref_ecma_string (str_p);
+  }
+
+  {
+    ecma_string_t *str_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+
+    ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+    ecma_string_t *result_p = ecma_stringbuilder_finalize (&builder);
+
+    TEST_ASSERT (ecma_compare_ecma_strings (result_p, str_p));
+  }
+
+  {
+    static const lit_utf8_byte_t string_data[] = "abc";
+
+    ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_A);
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_B);
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_LOWERCASE_C);
+    ecma_string_t *result_p = ecma_stringbuilder_finalize (&builder);
+
+    ecma_string_t *str_p = ecma_new_ecma_string_from_utf8 (string_data, sizeof (string_data) - 1);
+    TEST_ASSERT (ecma_compare_ecma_strings (result_p, str_p));
+    ecma_deref_ecma_string (result_p);
+    ecma_deref_ecma_string (str_p);
+  }
+
+  {
+    ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_1);
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_2);
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_3);
+    ecma_string_t *result_p = ecma_stringbuilder_finalize (&builder);
+
+    ecma_string_t *str_p = ecma_new_ecma_string_from_uint32 (123);
+    TEST_ASSERT (ecma_compare_ecma_strings (result_p, str_p));
+    ecma_deref_ecma_string (result_p);
+    ecma_deref_ecma_string (str_p);
+  }
+
+  {
+    static const lit_utf8_byte_t string_data[] = "abc";
+    ecma_string_t *uint_str_p = ecma_new_ecma_string_from_uint32 (234);
+
+    ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_1);
+    ecma_stringbuilder_append_raw (&builder, string_data, sizeof (string_data) - 1);
+    ecma_stringbuilder_append (&builder, uint_str_p);
+    ecma_stringbuilder_append_magic (&builder, LIT_MAGIC_STRING_STRING);
+    ecma_string_t *result_p = ecma_stringbuilder_finalize (&builder);
+
+    static const lit_utf8_byte_t expected_data[] = "1abc234string";
+    ecma_string_t *str_p = ecma_new_ecma_string_from_utf8 (expected_data, sizeof (expected_data) - 1);
+    TEST_ASSERT (ecma_compare_ecma_strings (result_p, str_p));
+    ecma_deref_ecma_string (result_p);
+    ecma_deref_ecma_string (str_p);
+  }
+
+  {
+    static const lit_utf8_byte_t string_data[] = "abc";
+    ecma_string_t *uint_str_p = ecma_new_ecma_string_from_uint32 (234);
+
+    ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_1);
+    ecma_stringbuilder_append_raw (&builder, string_data, sizeof (string_data) - 1);
+    ecma_stringbuilder_append (&builder, uint_str_p);
+    ecma_stringbuilder_append_magic (&builder, LIT_MAGIC_STRING_STRING);
+    ecma_stringbuilder_destroy (&builder);
```

Comment:
What does this case tests?

Reply:
This is just to test that we do not leak.